### PR TITLE
[PGEC-80] Refactoring marketplace touchpoints

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes
- Refactor touchpoints from enum 
- Add touchpoints collection for persistency

#### Motivation and Context

- Make touchpoint extensible

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/afm/marketplacebe/controller/TouchpointController.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    @Operation(summary = "Get touchpoint", security = {@SecurityRequirement(name = "ApiKey")}, tags = {"Touchpoint",})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = Touchpoint.class))),
+            @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "404", description = "Not Found", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "429", description = "Too many requests", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "500", description = "Service unavailable", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class)))})
+    @GetMapping(
+            value = "/touchpoint/{id}",
```
Comment (aomegax) [inline src/main/java/it/pagopa/afm/marketplacebe/controller/TouchpointController.java:63]: I think it is better to adopt the resource name at plural according to [richardson maturity model](https://martinfowler.com/articles/richardsonMaturityModel.html#level1)

**Review by aomegax — CHANGES_REQUESTED** [review]
👍 
only some cosmetic fix and re-generate swagger 😉

File: src/main/java/it/pagopa/afm/marketplacebe/controller/TouchpointController.java (hunk 1 above)
Comment (MarcoAGreco) [reply src/main/java/it/pagopa/afm/marketplacebe/controller/TouchpointController.java]: Nice point, updated with a0e9766 👍

File: src/main/java/it/pagopa/afm/marketplacebe/service/TouchpointService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    public Touchpoints getTouchpoints() {
+        List<Touchpoint> touchpointList = new ArrayList<>();
+
+        touchpointRepository
+                .findAll()
+                .forEach(t -> touchpointList.add(modelMapper.map(t, Touchpoint.class)));
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/afm/marketplacebe/service/TouchpointService.java:35]:
```suggestion
.findAll()
.map(elem-> modelMapper.map(t, Touchpoint.class))
Collectors.toList()
```

Comment (MarcoAGreco) [reply src/main/java/it/pagopa/afm/marketplacebe/service/TouchpointService.java:35]: I'd like to implement this suggestion, but `findAll()` method returns an Iterable, which does not provide a map method

Comment (jacopocarlini) [reply src/main/java/it/pagopa/afm/marketplacebe/service/TouchpointService.java:35]: Ah ok

**Review by jacopocarlini — APPROVED** [review]
Great 👍

File: openapi/openapi.json [hunk 3]
```diff
@@ -1,4 +1,12 @@
 {
+<<<<<<< HEAD
```
Comment (jacopocarlini) [inline openapi/openapi.json:2]: resolve the conflict

Comment (MarcoAGreco) [reply openapi/openapi.json]: Thanks! Solved with 6916209

**Review by jacopocarlini — APPROVED** [review]
OK 👍  
resolve the trivial code smells reported by sonar and the conflict before merge

**Review by aomegax — APPROVED** [review]
👍  but there are two unit test to fix

**Pull request merged**